File: app/webpacker/packs/customers/bonuses.js
import $ from 'jquery';

document.addEventListener('turbolinks:load', () => {
  const bonusSelector = $('#customer_bonus_original_bonus_id');
  const bonusDetailsWrapper = $('#bonus-details');
  const bonuses = $('#bonus-data').data('bonuses');

  function showBonusDetails(id) {
    const bonus = bonuses.find(el => parseInt(el.id, 10) === parseInt(id, 10));
    [
      'kind',
      'rollover_multiplier',
      'max_rollover_per_bet',
      'max_deposit_match',
      'min_odds_per_bet',
      'min_deposit',
      'valid_for_days',
      'percentage'
    ].map(key => $(`#customer_bonus_${key}`).val(bonus[key]));
    bonusDetailsWrapper.removeClass('d-none').show();
  }

  function hideBonusDetails() {
    bonusDetailsWrapper.hide();
  }

  bonusSelector.change(() => {
    const bonusId = bonusSelector.val();
    return bonusId ? showBonusDetails(bonusId) : hideBonusDetails()
  })
});

File: db/prime/bonuses.rb
puts 'Creating Bonuses ...'

Bonus.find_or_create_by!(code: 'bonus') do |bonus|
  bonus.kind = Bonus::DEPOSIT
  bonus.rollover_multiplier = 10
  bonus.max_rollover_per_bet = 200
  bonus.max_deposit_match = 200
  bonus.min_odds_per_bet = 1.5
  bonus.min_deposit = 10
  bonus.valid_for_days = 180
  bonus.expires_at = 3.months.from_now.end_of_month
  bonus.percentage = 100
  bonus.repeatable = true
end
